### PR TITLE
TEET-1224 fixes edge cases in the contract status and adds tests

### PR DESCRIPTION
**Pull request by jkorpelin**
Tests for contract statuses

### CODE REVIEW

File: app/backend/test/teet/contract/contract_db_test.clj [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                       [:thk.contract/procurement-id+procurement-part-id ["11111" "22222"]])]
+        (is (= (:thk.contract/status contract)
+               :thk.contract.status/deadline-overdue))))
+
+    (testing "Deadline is in the past so contract is in status deadline overdue"
+      (let [contract (contract-db/get-contract
+                       (tu/db)
+                       [:thk.contract/procurement-id+procurement-part-id ["11111" "22222"]])]
+        (is (= (:thk.contract/status contract)
+               :thk.contract.status/deadline-overdue))))
```
Comment (aleksievelizhanin) [inline app/backend/test/teet/contract/contract_db_test.clj:110]: Looks like duplicate to L98-L103 to me.

**Review by aleksievelizhanin — APPROVED** [review]
Looks legit, except one possible duplication

File: app/backend/test/teet/contract/contract_db_test.clj (hunk 1 above)
Comment (jkorpelin) [reply app/backend/test/teet/contract/contract_db_test.clj:110]: Good catch
